core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -90,30 +123,55 @@ public Argument build(Class<?> expectedType, T value, StatementContext ctx)
 
         // Enums must be bound as VARCHAR.
         if (expectedType.isEnum()) {
-            return new StringArgument(value.toString());
+            return STR_BUILDER.build(value.toString());
         }
 
         // Fallback to generic ObjectArgument
-        return new ObjectArgument(value);
+        return OBJ_BUILDER.build(value);
```

Comment:
This method has some redundant logic when `expectedType == Object.class`.

Have a look at this class in [#235](https://github.com/qualidafial/jdbi/blob/jdbi3-type-tokens/core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java) for a simpler flow.